src/jshint.js

```diff
@@ -2496,7 +2496,7 @@ var JSHINT = (function() {
     if (left) {
       if (left.type === "(identifier)") {
         if (left.value.match(/^[A-Z]([A-Z0-9_$]*[a-z][A-Za-z0-9_$]*)?$/)) {
-          if ("Number String Boolean Date Object Error Symbol".indexOf(left.value) === -1) {
+          if ("Array Number String Boolean Date Object Error Symbol".indexOf(left.value) === -1) {
```

Comment:
@caitp I don't think that's actually a problem:

```js
"Array Number String Boolean Date Object Error Symbol".indexOf("Number");
6
"Array Number String Boolean Date Object Error Symbol".indexOf("BigNumber");
-1
```

What could be a problem is truncated constructor names:

```js
"Array Number String Boolean Date Object Error Symbol".indexOf("umber");
7
```

...but this code path is only hit when the constructor begins with a capital letter. So while it's a bit of a hazard, [the option is deprecated](https://github.com/jshint/jshint/blob/7aa3e23e7e7bc53d1b19da25e669b30b6f75c5f5/src/options.js#L164-L183). I think this makes it safe to accept as-is. Can you confirm?